File: csharp/secret-handshake/SecretHandshake.cs
using System;
using System.Collections.Generic;

public static class SecretHandshake
{
    public static string[] Commands(int commandValue)
    {
        var commands = new Dictionary<int, string>(){
            { 1, "wink" },
            { 2, "double blink" },
            { 4, "close your eyes" },
            { 8, "jump" }
        };
        var res = new List<string>();
        foreach (KeyValuePair<int, string> entry in commands) {
            if ((entry.Key & commandValue) != 0) {
                res.Add(entry.Value);
            }
        }
        if ((commandValue & 16) != 0) {
            res.Reverse();
        }
        return res.ToArray();
    }
}

File: typescript/secret-handshake/secret-handshake.ts
const COMMANDS: [number, string][] = [
  [0b1, 'wink'],
  [0b10, 'double blink'],
  [0b100, 'close your eyes'],
  [0b1000, 'jump'],
  [0b10000, 'reverse']
]

class HandShake {
  private n: number

  constructor(n: number) {
    this.n = n
  }

  commands = () => {
    const res: string[] = []
    COMMANDS.forEach((commandTuple) => {
      const [bit, command] = commandTuple
      if ((this.n & bit) === bit) {
        if (bit === 0b10000) {
          res.reverse();
        } else {
          res.push(command)
        }
      }
    })
    return res
  }
}

export default HandShake